kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/BrokerConfigsResource.java

```diff
@@ -0,0 +1,124 @@
+
+  private final BrokerConfigManager brokerConfigManager;
+  private final CrnFactory crnFactory;
+  private final UrlFactory urlFactory;
+
+  public BrokerConfigsResource(
+      BrokerConfigManager brokerConfigManager,
+      CrnFactory crnFactory,
+      UrlFactory urlFactory) {
+    this.brokerConfigManager = Objects.requireNonNull(brokerConfigManager);
+    this.crnFactory = Objects.requireNonNull(crnFactory);
+    this.urlFactory = Objects.requireNonNull(urlFactory);
+  }
+
+  @GET
+  @Produces(Versions.JSON_API)
+  public void listBrokerConfigs(
+      @Suspended AsyncResponse asyncResponse,
+      @PathParam("clusterId") String clusterId,
+      @PathParam("brokerId") int brokerId) {
+    CompletableFuture<ListBrokerConfigsResponse> response =
+        brokerConfigManager.listBrokerConfigs(clusterId, brokerId)
+            .thenApply(
+                configs ->
+                    new ListBrokerConfigsResponse(
+                        new CollectionLink(
+                            urlFactory.create(
+                                "v3", "clusters", clusterId, "brokers", String.valueOf(brokerId),
+                                "configs"),
+                            /* next= */null),
+                        configs.stream()
+                            .sorted(Comparator.comparing(BrokerConfig::getName))
+                            .map(this::toBrokerConfigData)
+                            .collect(Collectors.toList())));
+    AsyncResponses.asyncResume(asyncResponse, response);
+  }
+
+  @GET
+  @Path("/{name}")
+  @Produces(Versions.JSON_API)
+  public void getBrokerConfig(
+      @Suspended AsyncResponse asyncResponse,
+      @PathParam("clusterId") String clusterId,
+      @PathParam("brokerId") int brokerId,
+      @PathParam("name") String name
+  ) {
+    CompletableFuture<GetBrokerConfigResponse> response =
+        brokerConfigManager.getBrokerConfig(clusterId, brokerId, name)
+            .thenApply(broker -> broker.orElseThrow(NotFoundException::new))
+            .thenApply(broker -> new GetBrokerConfigResponse(toBrokerConfigData(broker)));
+
+    AsyncResponses.asyncResume(asyncResponse, response);
+  }
+
+  private BrokerConfigData toBrokerConfigData(BrokerConfig brokerConfig) {
+    return new BrokerConfigData(
+        crnFactory.create(
+            ClusterData.ELEMENT_TYPE,
+            brokerConfig.getClusterId(),
+            BrokerData.ELEMENT_TYPE,
+            String.valueOf(brokerConfig.getBrokerId()),
+            BrokerConfigData.ELEMENT_TYPE,
+            brokerConfig.getName()),
+        new ResourceLink(
+            urlFactory.create(
+                "v3",
+                "clusters",
+                brokerConfig.getClusterId(),
+                "brokers",
+                String.valueOf(brokerConfig.getBrokerId()),
+                "configs",
+                brokerConfig.getName())),
+        brokerConfig.getClusterId(),
+        String.valueOf(brokerConfig.getBrokerId()),
```

Comment:
I think here is `brokerConfig.getBrokerId()`, without the `String.valueOf`.

Reply:
The create method has var args as `(String ... components), so converted the int brokerId to String.

Reply:
Update - modified when being passed to `BrokerConfigData`. I thought this was about creating url.